Mathlib/Algebra/Module/Submodule/Map.lean

```diff
@@ -619,3 +619,19 @@ def compatibleMaps : Submodule R (N →ₗ[R] N₂) where
 #align submodule.compatible_maps Submodule.compatibleMaps
 
 end Submodule
+
+namespace LinearMap
+
+variable [Semiring R] [AddCommMonoid M] [AddCommMonoid M₁] [Module R M] [Module R M₁]
+
+/-- A linear map between two modules restricts to a linear map from any submodule p of the
+domain onto the image of that submodule.
+
+This is the linear version of `AddMonoidHom.addSubmonoidMap` and `AddMonoidHom.addSubgroupMap`.-/
+def submoduleMap (f : M →ₗ[R] M₁) (p : Submodule R M) : p →ₗ[R] p.map f :=
```

Comment:
Can you add a (maybe simp?) lemma `↑(f.submoduleMap p x) = f x`? Thanks.

Reply:
Added, and I did indeed tag it `@[simp]` since `simp` doesn't unfold the definition and thus doesn't use `restrict_coe_apply`